ibis/expr/operations.py

```diff
@@ -1025,6 +1025,17 @@ def __init__(self, expr, window):
         if table is not None:
             window = window.bind(table)
 
+        if window.max_lookback is not None:
+            if len(window._order_by) != 1:
+                raise com.IbisInputError(
+                    "'max lookback windows must be ordered by time"
+                )
+            order_var = window._order_by[0].op().args[0]
+            if not isinstance(order_var._dtype, dt.Timestamp):
```

Comment:
You can use the public API here: `.type()` to get the `dt.DataType` instance from an expression.

Reply:
Fixed in next commit